commands/remove/index.js

```diff
@@ -0,0 +1,149 @@
+"use strict";
+
+const path = require("path");
+const pMap = require("p-map");
+const fs = require("fs-extra");
+
+const Command = require("@lerna/command");
+const { getFilteredPackages } = require("@lerna/filter-options");
+const npmUninstall = require("@lerna/npm-uninstall");
+
+module.exports = factory;
+
+function factory(argv) {
+  return new RemoveCommand(argv);
+}
+
+class RemoveCommand extends Command {
+  get requiresGit() {
+    return false;
+  }
+
+  initialize() {
+    this.dependencyToRemove = this.options.pkg;
+
+    this.dirs = new Set(this.options.globs.map(fp => path.resolve(this.project.rootPath, fp)));
+
+    let chain = Promise.resolve();
+
+    chain = chain.then(() => getFilteredPackages(this.packageGraph, this.execOpts, this.options));
+    chain = chain.then(filteredPackages => {
+      this.filteredPackages = filteredPackages;
+    });
+
+    chain = chain.then(() => this.collectPackagesToChange());
+    chain = chain.then(packagesToChange => {
+      this.packagesToChange = packagesToChange;
+    });
+
+    return chain.then(() => {
+      const proceed = this.packagesToChange.length > 0;
+
+      if (!proceed) {
+        this.logger.warn(`No packages found where ${this.dependencyToRemove} can be removed.`);
+      }
+
+      return proceed;
+    });
+  }
+
+  execute() {
+    const numberOfPackages = `${this.packagesToChange.length} package${
+      this.packagesToChange.length > 1 ? "s" : ""
+    }`;
+
+    this.logger.info("", `Remove ${this.dependencyToRemove} from ${numberOfPackages}`);
+
+    let chain = Promise.resolve();
+
+    chain = chain.then(() => this.removePackage());
+
+    // if (this.options.bootstrap !== false) {
+    //   chain = chain.then(() => {
+    //     const argv = Object.assign({}, this.options, {
+    //       args: [],
+    //       cwd: this.project.rootPath,
+    //       // silence initial cli version logging, etc
+    //       composed: "add",
```

Comment:
We should definitely still be doing this, instead of calling `npm uninstall`. Ensuring that `ci: false` is also passed makes sure we run `npm install` which will synchronize the lockfile and `node_modules` appropriately?